pkg/operator/ceph/cluster/newosd/osd.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package newosd
```

Comment:
Is this `newosd` package temporary, or you see it as a clean start and should be a separate package?

Reply:
I've been viewing it as a clean start and a separate package. I don't want to add even more clutter to the existing code. At some point, the existing `osd` would be replaced entirely by `newosd`, and then this would be renamed to `osd`.

Reply:
I wonder if we can achieve your goal by creating new source files in the same package. We will need to support the legacy OSDs for some time so it will take a long time for the old package to go away. It feels like it would be more complicated to figure out what to copy over instead of finding a way to cleanup the existing package.